### PR TITLE
chore: add scripts for preview and building subsections of the wiki

### LINKED ISSUE #1254: Improve setup by using cli and splitting up the wiki
**Is your feature request related to a problem? Please describe.**

The idea to improve build time for local development and even remote workflows is to split up the wiki into multiple parts and only run/build the parts you need. 

**Describe the solution you'd like**

For this we would like to use the cli. It could ask you which parts you want to run like `Identity`, `Get Started`, `Build` and so on. It should be possible to do that over a nice TUI and through flags like `iota-wiki start identity`
This would improve the local development experience.
Maybe we could even use this to detect in a PR which part of the Wiki changed to only run workflows on that part (might be a problem for broken links, so do we even want to do this?)

### PR DESCRIPTION
**Pull request by VmMad**
# Description of change

Handle selection of sections of the wiki by splitting the config and selecting it with an environment variable.
Add scripts to start a subsection of the wiki directly: `yarn start:section:{section}`
Only show the selected section in the navbar

## Links to any relevant issues

Fixes #1254 

## Type of change

- Enhancement

## Change checklist

- [x] I have followed the [contribution guidelines](https://github.com/iota-wiki/iota-wiki/blob/main/.github/CONTRIBUTING.md) for this project
- [x] I have performed a self-review of my own changes
- [ ] I have made sure that added/changed links still work
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,7 +19,15 @@
     "prepare": "yarn build:plugins && yarn build:theme && yarn checkout",
     "docusaurus": "docusaurus",
     "start": "NODE_OPTIONS='--max-old-space-size=4096' docusaurus start",
+    "start:section:build": "NODE_OPTIONS='--max-old-space-size=4096' SELECTED_SECTION='build' docusaurus start",
+    "start:section:get-started": "NODE_OPTIONS='--max-old-space-size=4096' SELECTED_SECTION='get-started' docusaurus start",
+    "start:section:learn": "NODE_OPTIONS='--max-old-space-size=4096' SELECTED_SECTION='learn' docusaurus start",
+    "start:section:maintain": "NODE_OPTIONS='--max-old-space-size=4096' SELECTED_SECTION='maintain' docusaurus start",
```
Comment (jlvandenhout) [inline package.json:25]:
```suggestion
    "start:section:build": "SELECTED_SECTION='build' yarn start",
    "start:section:get-started": "SELECTED_SECTION='get-started' yarn start",
    "start:section:learn": "SELECTED_SECTION='learn' yarn start",
    "start:section:maintain": "SELECTED_SECTION='maintain' yarn start",
```

**Review by jlvandenhout — CHANGES_REQUESTED** [review]

**Comment by VmMad** [comment]
Comment for the reviewers:
I added the build scripts to this PR too, but they won't work unless the build configuration is updated. I didn't want to mess with your plugin configuration so if you want me to check this let me know

**Comment by jlvandenhout** [comment]
I guess it's the redirect plugin that is complaining? So we'd need to do something similar to what you did in `contentPlugins.js`, but then in `articleRedirects.js` and the list of redirects in the `@docusaurus/plugin-client-redirects` plugin config I guess? I think it's fine to do that in this PR too 👍

**Comment by Dr-Electron** [comment]
> I guess it's the redirect plugin that is complaining? So we'd need to do something similar to what you did in `contentPlugins.js`, but then in `articleRedirects.js` and the list of redirects in the `@docusaurus/plugin-client-redirects` plugin config I guess? I think it's fine to do that in this PR too 👍

I guess we could simplify that by not allowing builds to be split and in start we don't care about redirects? 🤔 
Do we need to build a single section? Wdyt @jlvandenhout

**Comment by jlvandenhout** [comment]
@Dr-Electron That's true, so let's remove the build scripts.

File: package.json [hunk 2]
```diff
@@ -19,7 +19,15 @@
...
     "start": "NODE_OPTIONS='--max-old-space-size=4096' docusaurus start",
+    "start:section:build": "SELECTED_SECTION='build' yarn start",
+    "start:section:get-started": "SELECTED_SECTION='get-started' yarn start",
+    "start:section:learn": "SELECTED_SECTION='learn' yarn start",
+    "start:section:maintain": "SELECTED_SECTION='maintain' yarn start",
     "build": "NODE_OPTIONS='--max-old-space-size=4096' docusaurus build",
+    "build:section:build": "SELECTED_SECTION='build' yarn build",
+    "build:section:get-started": "SELECTED_SECTION='get-started' yarn build",
+    "build:section:learn": "SELECTED_SECTION='learn' yarn build",
+    "build:section:maintain": "SELECTED_SECTION='maintain' yarn build",
```
Comment (jlvandenhout) [inline package.json:30]: We can remove these, because we will not use them for now anyways.

```suggestion
```

**Review by jlvandenhout — CHANGES_REQUESTED** [review]

**Review by jlvandenhout — DISMISSED** [review]

**Review by Dr-Electron — APPROVED** [review]

**Review by jlvandenhout — APPROVED** [review]

**Pull request merged**